src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationEventSource.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Diagnostics.Tracing;
+
+namespace System.Xml.Serialization
+{
+    [EventSource(Name = "System.Xml.Serialzation.XmlSerialization")]
+    internal class XmlSerializationEventSource : EventSource
+    {
+        internal static XmlSerializationEventSource Log = new XmlSerializationEventSource();
+
+        [Event(EventIds.XmlSerializerExpired, Level = EventLevel.Informational, 
+            Message = "Pre-generated serializer '{0}' has expired. You need to re-generate serializer for '{1}'")]
+        internal void XmlSerializerExpired(string serializerName, string type)
```

Comment:
I've discovered there is a way to keep this in the resource file. You use the LocalizationResources property of the EventSource attribute. See [here](https://github.com/dotnet/coreclr/blob/76ba8d518eb5830988878fca01e4a6096c1b0b2a/src/mscorlib/src/System/Threading/Tasks/TPLETWProvider.cs#L27) for an example. [Here](https://github.com/dotnet/coreclr/blob/79b8f94cc52b6b79945074b748508a0b3cbb8be6/src/System.Private.CoreLib/Resources/Strings.resx#L1996) is one of the corresponding lines in the resx file. So in this case the resource name used in the resx file would be event_XmlSerializerExpired. I think the name you would want to specify for the LocalizationResources property will be `"FxResources.System.Private.Xml.SR"` but you will need to verify this.